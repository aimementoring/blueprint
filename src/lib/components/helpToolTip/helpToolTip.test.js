import React from 'react';
import renderer from 'react-test-renderer';
import HelpToolTip from './helpToolTip.js';

describe('HelpToolTip', () => {
  it('renders with children', () => {
    const tree = renderer
      .create(
        <HelpToolTip>
          <h2>A headline</h2>
          <p>Some children with content</p>
        </HelpToolTip>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses placement and trigger argument', () => {
    const tree = renderer
      .create(
        <HelpToolTip placement="bottom" trigger="click" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses theme', () => {
    const tree = renderer
      .create(<HelpToolTip theme="purple" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
