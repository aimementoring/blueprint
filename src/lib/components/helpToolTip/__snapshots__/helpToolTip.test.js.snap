// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpToolTip renders with children 1`] = `
<div
  className="theme-base"
>
  <span
    className="trigger trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      className="questionMarkIcon"
      fill="none"
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 10C17.3291 10 14.8175 11.0403 12.9291 12.9291C11.0403 14.8175 10 17.3291 10 20C10 22.6709 11.0403 25.1825 12.9291 27.0709C14.8175 28.9597 17.3291 30 20 30C22.6709 30 25.1825 28.9597 27.0709 27.0709C28.9597 25.1825 30 22.6709 30 20C30 17.3291 28.9597 14.8175 27.0709 12.9291C25.1825 11.0403 22.6709 10 20 10ZM20 28.5996C15.2582 28.5996 11.4004 24.7418 11.4004 20C11.4004 15.2582 15.2582 11.4004 20 11.4004C24.7418 11.4004 28.5996 15.2582 28.5996 20C28.5996 24.7418 24.7418 28.5996 20 28.5996Z"
        fill="#5E6366"
      />
      <path
        d="M22.9492 14.626C22.1846 13.8615 21.1728 13.426 20.1002 13.4011C18.9654 13.3744 17.8936 13.7958 17.0827 14.588C16.2713 15.3805 15.8245 16.4414 15.8245 17.5748C15.8245 17.9617 16.1378 18.2751 16.5247 18.2751C16.9116 18.2751 17.2249 17.9617 17.2249 17.5748C17.2249 16.8212 17.5221 16.1162 18.0612 15.5897C18.6 15.0632 19.3124 14.7836 20.0674 14.8011C21.5256 14.8352 22.7396 16.0497 22.7741 17.5079C22.7982 18.5407 22.2533 19.4952 21.3518 19.9993C20.0862 20.707 19.2997 22.03 19.2997 23.4562C19.2997 23.8431 19.6131 24.1547 20 24.1547C20.3868 24.1547 20.7002 23.8392 20.7002 23.4527C20.7002 22.5372 21.2118 21.6825 22.035 21.2221C23.3908 20.4641 24.2105 19.0282 24.1737 17.4755C24.1488 16.4024 23.7137 15.3906 22.9492 14.626Z"
        fill="#5E6366"
      />
      <path
        d="M20.6867 25.7626C20.6819 25.7402 20.6762 25.7179 20.6696 25.696C20.663 25.6742 20.6552 25.6527 20.6464 25.6317C20.6377 25.6107 20.628 25.5897 20.6175 25.5696C20.6066 25.5494 20.5948 25.5297 20.5821 25.5109C20.5694 25.4917 20.5558 25.4733 20.5414 25.4553C20.5269 25.4378 20.5112 25.4203 20.495 25.4041C20.4788 25.3884 20.4617 25.3726 20.4442 25.3582C20.4263 25.3437 20.4079 25.3297 20.3886 25.3175C20.3698 25.3048 20.3501 25.293 20.33 25.282C20.3099 25.2715 20.2889 25.2615 20.2679 25.2527C20.2469 25.2444 20.225 25.2365 20.2035 25.23C20.1816 25.223 20.1593 25.2173 20.137 25.2129C20.1142 25.2085 20.0915 25.205 20.0687 25.2028C20.0232 25.198 19.9768 25.198 19.9313 25.2028C19.9086 25.205 19.8858 25.2085 19.863 25.2129C19.8407 25.2173 19.8184 25.223 19.7965 25.23C19.7751 25.2365 19.7532 25.2444 19.7322 25.2527C19.7112 25.2615 19.6902 25.2715 19.67 25.282C19.6499 25.293 19.6302 25.3048 19.6114 25.3175C19.5921 25.3297 19.5738 25.3437 19.5558 25.3582C19.5383 25.3726 19.5212 25.3884 19.5051 25.4041C19.4889 25.4203 19.4731 25.4378 19.4587 25.4553C19.4442 25.4733 19.4307 25.4917 19.418 25.5109C19.4053 25.5297 19.3935 25.5494 19.3825 25.5696C19.372 25.5897 19.3619 25.6107 19.3536 25.6317C19.3449 25.6527 19.337 25.6742 19.3304 25.696C19.3234 25.7179 19.3182 25.7402 19.3134 25.7626C19.309 25.7849 19.3055 25.8081 19.3033 25.8304C19.3011 25.8536 19.2998 25.8764 19.2998 25.8995C19.2998 25.9223 19.3011 25.9455 19.3033 25.9683C19.3055 25.991 19.309 26.0138 19.3134 26.0361C19.3182 26.0584 19.3234 26.0807 19.3304 26.1026C19.337 26.1245 19.3449 26.1464 19.3536 26.1674C19.3619 26.1884 19.372 26.2094 19.3825 26.2291C19.3935 26.2497 19.4053 26.2693 19.418 26.2882C19.4307 26.3074 19.4442 26.3258 19.4587 26.3433C19.4731 26.3613 19.4889 26.3783 19.5051 26.3945C19.5212 26.4107 19.5383 26.4265 19.5558 26.4409C19.5738 26.4553 19.5921 26.4689 19.6114 26.4816C19.6302 26.4943 19.6499 26.5061 19.67 26.5166C19.6902 26.5276 19.7112 26.5372 19.7322 26.5459C19.7532 26.5547 19.7751 26.5626 19.7965 26.5691C19.8184 26.5757 19.8407 26.5814 19.863 26.5862C19.8858 26.5906 19.9086 26.5941 19.9313 26.5963C19.9541 26.5985 19.9773 26.5998 20 26.5998C20.0228 26.5998 20.046 26.5985 20.0687 26.5963C20.0915 26.5941 20.1142 26.5906 20.137 26.5862C20.1593 26.5814 20.1816 26.5757 20.2035 26.5691C20.225 26.5626 20.2469 26.5547 20.2679 26.5459C20.2889 26.5372 20.3099 26.5276 20.3296 26.5166C20.3497 26.5061 20.3698 26.4943 20.3886 26.4816C20.4079 26.4689 20.4263 26.4553 20.4438 26.4409C20.4617 26.4265 20.4788 26.4107 20.495 26.3945C20.5112 26.3783 20.5269 26.3613 20.5414 26.3433C20.5558 26.3258 20.5694 26.3074 20.5821 26.2882C20.5948 26.2693 20.6066 26.2497 20.6175 26.2291C20.628 26.2094 20.6377 26.1884 20.6464 26.1674C20.6552 26.1464 20.663 26.1245 20.6696 26.1026C20.6762 26.0807 20.6819 26.0584 20.6867 26.0361C20.6911 26.0138 20.6946 25.991 20.6967 25.9683C20.6989 25.9455 20.7002 25.9223 20.7002 25.8995C20.7002 25.8764 20.6989 25.8536 20.6967 25.8304C20.6946 25.8081 20.6911 25.7849 20.6867 25.7626Z"
        fill="#5E6366"
      />
    </svg>
  </span>
</div>
`;

exports[`HelpToolTip uses placement and trigger argument 1`] = `
<div
  className="theme-base"
>
  <span
    className="trigger trigger"
    onClick={[Function]}
    onTouchEnd={[Function]}
  >
    <svg
      className="questionMarkIcon"
      fill="none"
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 10C17.3291 10 14.8175 11.0403 12.9291 12.9291C11.0403 14.8175 10 17.3291 10 20C10 22.6709 11.0403 25.1825 12.9291 27.0709C14.8175 28.9597 17.3291 30 20 30C22.6709 30 25.1825 28.9597 27.0709 27.0709C28.9597 25.1825 30 22.6709 30 20C30 17.3291 28.9597 14.8175 27.0709 12.9291C25.1825 11.0403 22.6709 10 20 10ZM20 28.5996C15.2582 28.5996 11.4004 24.7418 11.4004 20C11.4004 15.2582 15.2582 11.4004 20 11.4004C24.7418 11.4004 28.5996 15.2582 28.5996 20C28.5996 24.7418 24.7418 28.5996 20 28.5996Z"
        fill="#5E6366"
      />
      <path
        d="M22.9492 14.626C22.1846 13.8615 21.1728 13.426 20.1002 13.4011C18.9654 13.3744 17.8936 13.7958 17.0827 14.588C16.2713 15.3805 15.8245 16.4414 15.8245 17.5748C15.8245 17.9617 16.1378 18.2751 16.5247 18.2751C16.9116 18.2751 17.2249 17.9617 17.2249 17.5748C17.2249 16.8212 17.5221 16.1162 18.0612 15.5897C18.6 15.0632 19.3124 14.7836 20.0674 14.8011C21.5256 14.8352 22.7396 16.0497 22.7741 17.5079C22.7982 18.5407 22.2533 19.4952 21.3518 19.9993C20.0862 20.707 19.2997 22.03 19.2997 23.4562C19.2997 23.8431 19.6131 24.1547 20 24.1547C20.3868 24.1547 20.7002 23.8392 20.7002 23.4527C20.7002 22.5372 21.2118 21.6825 22.035 21.2221C23.3908 20.4641 24.2105 19.0282 24.1737 17.4755C24.1488 16.4024 23.7137 15.3906 22.9492 14.626Z"
        fill="#5E6366"
      />
      <path
        d="M20.6867 25.7626C20.6819 25.7402 20.6762 25.7179 20.6696 25.696C20.663 25.6742 20.6552 25.6527 20.6464 25.6317C20.6377 25.6107 20.628 25.5897 20.6175 25.5696C20.6066 25.5494 20.5948 25.5297 20.5821 25.5109C20.5694 25.4917 20.5558 25.4733 20.5414 25.4553C20.5269 25.4378 20.5112 25.4203 20.495 25.4041C20.4788 25.3884 20.4617 25.3726 20.4442 25.3582C20.4263 25.3437 20.4079 25.3297 20.3886 25.3175C20.3698 25.3048 20.3501 25.293 20.33 25.282C20.3099 25.2715 20.2889 25.2615 20.2679 25.2527C20.2469 25.2444 20.225 25.2365 20.2035 25.23C20.1816 25.223 20.1593 25.2173 20.137 25.2129C20.1142 25.2085 20.0915 25.205 20.0687 25.2028C20.0232 25.198 19.9768 25.198 19.9313 25.2028C19.9086 25.205 19.8858 25.2085 19.863 25.2129C19.8407 25.2173 19.8184 25.223 19.7965 25.23C19.7751 25.2365 19.7532 25.2444 19.7322 25.2527C19.7112 25.2615 19.6902 25.2715 19.67 25.282C19.6499 25.293 19.6302 25.3048 19.6114 25.3175C19.5921 25.3297 19.5738 25.3437 19.5558 25.3582C19.5383 25.3726 19.5212 25.3884 19.5051 25.4041C19.4889 25.4203 19.4731 25.4378 19.4587 25.4553C19.4442 25.4733 19.4307 25.4917 19.418 25.5109C19.4053 25.5297 19.3935 25.5494 19.3825 25.5696C19.372 25.5897 19.3619 25.6107 19.3536 25.6317C19.3449 25.6527 19.337 25.6742 19.3304 25.696C19.3234 25.7179 19.3182 25.7402 19.3134 25.7626C19.309 25.7849 19.3055 25.8081 19.3033 25.8304C19.3011 25.8536 19.2998 25.8764 19.2998 25.8995C19.2998 25.9223 19.3011 25.9455 19.3033 25.9683C19.3055 25.991 19.309 26.0138 19.3134 26.0361C19.3182 26.0584 19.3234 26.0807 19.3304 26.1026C19.337 26.1245 19.3449 26.1464 19.3536 26.1674C19.3619 26.1884 19.372 26.2094 19.3825 26.2291C19.3935 26.2497 19.4053 26.2693 19.418 26.2882C19.4307 26.3074 19.4442 26.3258 19.4587 26.3433C19.4731 26.3613 19.4889 26.3783 19.5051 26.3945C19.5212 26.4107 19.5383 26.4265 19.5558 26.4409C19.5738 26.4553 19.5921 26.4689 19.6114 26.4816C19.6302 26.4943 19.6499 26.5061 19.67 26.5166C19.6902 26.5276 19.7112 26.5372 19.7322 26.5459C19.7532 26.5547 19.7751 26.5626 19.7965 26.5691C19.8184 26.5757 19.8407 26.5814 19.863 26.5862C19.8858 26.5906 19.9086 26.5941 19.9313 26.5963C19.9541 26.5985 19.9773 26.5998 20 26.5998C20.0228 26.5998 20.046 26.5985 20.0687 26.5963C20.0915 26.5941 20.1142 26.5906 20.137 26.5862C20.1593 26.5814 20.1816 26.5757 20.2035 26.5691C20.225 26.5626 20.2469 26.5547 20.2679 26.5459C20.2889 26.5372 20.3099 26.5276 20.3296 26.5166C20.3497 26.5061 20.3698 26.4943 20.3886 26.4816C20.4079 26.4689 20.4263 26.4553 20.4438 26.4409C20.4617 26.4265 20.4788 26.4107 20.495 26.3945C20.5112 26.3783 20.5269 26.3613 20.5414 26.3433C20.5558 26.3258 20.5694 26.3074 20.5821 26.2882C20.5948 26.2693 20.6066 26.2497 20.6175 26.2291C20.628 26.2094 20.6377 26.1884 20.6464 26.1674C20.6552 26.1464 20.663 26.1245 20.6696 26.1026C20.6762 26.0807 20.6819 26.0584 20.6867 26.0361C20.6911 26.0138 20.6946 25.991 20.6967 25.9683C20.6989 25.9455 20.7002 25.9223 20.7002 25.8995C20.7002 25.8764 20.6989 25.8536 20.6967 25.8304C20.6946 25.8081 20.6911 25.7849 20.6867 25.7626Z"
        fill="#5E6366"
      />
    </svg>
  </span>
</div>
`;

exports[`HelpToolTip uses theme 1`] = `
<div
  className="theme-purple"
>
  <span
    className="trigger trigger"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      className="questionMarkIcon"
      fill="none"
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 10C17.3291 10 14.8175 11.0403 12.9291 12.9291C11.0403 14.8175 10 17.3291 10 20C10 22.6709 11.0403 25.1825 12.9291 27.0709C14.8175 28.9597 17.3291 30 20 30C22.6709 30 25.1825 28.9597 27.0709 27.0709C28.9597 25.1825 30 22.6709 30 20C30 17.3291 28.9597 14.8175 27.0709 12.9291C25.1825 11.0403 22.6709 10 20 10ZM20 28.5996C15.2582 28.5996 11.4004 24.7418 11.4004 20C11.4004 15.2582 15.2582 11.4004 20 11.4004C24.7418 11.4004 28.5996 15.2582 28.5996 20C28.5996 24.7418 24.7418 28.5996 20 28.5996Z"
        fill="#5E6366"
      />
      <path
        d="M22.9492 14.626C22.1846 13.8615 21.1728 13.426 20.1002 13.4011C18.9654 13.3744 17.8936 13.7958 17.0827 14.588C16.2713 15.3805 15.8245 16.4414 15.8245 17.5748C15.8245 17.9617 16.1378 18.2751 16.5247 18.2751C16.9116 18.2751 17.2249 17.9617 17.2249 17.5748C17.2249 16.8212 17.5221 16.1162 18.0612 15.5897C18.6 15.0632 19.3124 14.7836 20.0674 14.8011C21.5256 14.8352 22.7396 16.0497 22.7741 17.5079C22.7982 18.5407 22.2533 19.4952 21.3518 19.9993C20.0862 20.707 19.2997 22.03 19.2997 23.4562C19.2997 23.8431 19.6131 24.1547 20 24.1547C20.3868 24.1547 20.7002 23.8392 20.7002 23.4527C20.7002 22.5372 21.2118 21.6825 22.035 21.2221C23.3908 20.4641 24.2105 19.0282 24.1737 17.4755C24.1488 16.4024 23.7137 15.3906 22.9492 14.626Z"
        fill="#5E6366"
      />
      <path
        d="M20.6867 25.7626C20.6819 25.7402 20.6762 25.7179 20.6696 25.696C20.663 25.6742 20.6552 25.6527 20.6464 25.6317C20.6377 25.6107 20.628 25.5897 20.6175 25.5696C20.6066 25.5494 20.5948 25.5297 20.5821 25.5109C20.5694 25.4917 20.5558 25.4733 20.5414 25.4553C20.5269 25.4378 20.5112 25.4203 20.495 25.4041C20.4788 25.3884 20.4617 25.3726 20.4442 25.3582C20.4263 25.3437 20.4079 25.3297 20.3886 25.3175C20.3698 25.3048 20.3501 25.293 20.33 25.282C20.3099 25.2715 20.2889 25.2615 20.2679 25.2527C20.2469 25.2444 20.225 25.2365 20.2035 25.23C20.1816 25.223 20.1593 25.2173 20.137 25.2129C20.1142 25.2085 20.0915 25.205 20.0687 25.2028C20.0232 25.198 19.9768 25.198 19.9313 25.2028C19.9086 25.205 19.8858 25.2085 19.863 25.2129C19.8407 25.2173 19.8184 25.223 19.7965 25.23C19.7751 25.2365 19.7532 25.2444 19.7322 25.2527C19.7112 25.2615 19.6902 25.2715 19.67 25.282C19.6499 25.293 19.6302 25.3048 19.6114 25.3175C19.5921 25.3297 19.5738 25.3437 19.5558 25.3582C19.5383 25.3726 19.5212 25.3884 19.5051 25.4041C19.4889 25.4203 19.4731 25.4378 19.4587 25.4553C19.4442 25.4733 19.4307 25.4917 19.418 25.5109C19.4053 25.5297 19.3935 25.5494 19.3825 25.5696C19.372 25.5897 19.3619 25.6107 19.3536 25.6317C19.3449 25.6527 19.337 25.6742 19.3304 25.696C19.3234 25.7179 19.3182 25.7402 19.3134 25.7626C19.309 25.7849 19.3055 25.8081 19.3033 25.8304C19.3011 25.8536 19.2998 25.8764 19.2998 25.8995C19.2998 25.9223 19.3011 25.9455 19.3033 25.9683C19.3055 25.991 19.309 26.0138 19.3134 26.0361C19.3182 26.0584 19.3234 26.0807 19.3304 26.1026C19.337 26.1245 19.3449 26.1464 19.3536 26.1674C19.3619 26.1884 19.372 26.2094 19.3825 26.2291C19.3935 26.2497 19.4053 26.2693 19.418 26.2882C19.4307 26.3074 19.4442 26.3258 19.4587 26.3433C19.4731 26.3613 19.4889 26.3783 19.5051 26.3945C19.5212 26.4107 19.5383 26.4265 19.5558 26.4409C19.5738 26.4553 19.5921 26.4689 19.6114 26.4816C19.6302 26.4943 19.6499 26.5061 19.67 26.5166C19.6902 26.5276 19.7112 26.5372 19.7322 26.5459C19.7532 26.5547 19.7751 26.5626 19.7965 26.5691C19.8184 26.5757 19.8407 26.5814 19.863 26.5862C19.8858 26.5906 19.9086 26.5941 19.9313 26.5963C19.9541 26.5985 19.9773 26.5998 20 26.5998C20.0228 26.5998 20.046 26.5985 20.0687 26.5963C20.0915 26.5941 20.1142 26.5906 20.137 26.5862C20.1593 26.5814 20.1816 26.5757 20.2035 26.5691C20.225 26.5626 20.2469 26.5547 20.2679 26.5459C20.2889 26.5372 20.3099 26.5276 20.3296 26.5166C20.3497 26.5061 20.3698 26.4943 20.3886 26.4816C20.4079 26.4689 20.4263 26.4553 20.4438 26.4409C20.4617 26.4265 20.4788 26.4107 20.495 26.3945C20.5112 26.3783 20.5269 26.3613 20.5414 26.3433C20.5558 26.3258 20.5694 26.3074 20.5821 26.2882C20.5948 26.2693 20.6066 26.2497 20.6175 26.2291C20.628 26.2094 20.6377 26.1884 20.6464 26.1674C20.6552 26.1464 20.663 26.1245 20.6696 26.1026C20.6762 26.0807 20.6819 26.0584 20.6867 26.0361C20.6911 26.0138 20.6946 25.991 20.6967 25.9683C20.6989 25.9455 20.7002 25.9223 20.7002 25.8995C20.7002 25.8764 20.6989 25.8536 20.6967 25.8304C20.6946 25.8081 20.6911 25.7849 20.6867 25.7626Z"
        fill="#5E6366"
      />
    </svg>
  </span>
</div>
`;
